timeserio/preprocessing/datetime.py

```diff
@@ -216,7 +217,56 @@ def transformed_columns(self, input_columns):
         return input_columns | set(self.attributes_)
 
 
-class LagFeaturizer(BaseEstimator, TransformerMixin, CallableMixin):
+class _BaseLagFeaturizer(BaseEstimator, TransformerMixin, CallableMixin):
+    _datetime_column: str
+    _columns: List[str]
+    _lags: List
```

Comment:
If `lags` are used in the base class, e.g. in the `transformed_columns`, it should probably be set in the __init__ here too?